client/src/test/java/io/cloudsoft/winrm4j/client/WinRmClientRecordedTest.java

```diff
@@ -0,0 +1,281 @@
+        try (MockWebServer server = new MockWebServer()) {
+            // Use to fix the port for network capture
+            // server.start(5985);
+            URL serverUrl = server.url("/wsman").url();
+            RecordedSessionDispatcher dispatcher = new RecordedSessionDispatcher(recordingName, serverUrl.toString());
+            server.setDispatcher(dispatcher);
+            doRequest(serverUrl);
+            assertFalse(dispatcher.hasErrors(), "Dispatcher reported errors, see logs for details");
+        }
+    }
+
+    private void doRequest(URL url) {
+        WinRmClient.Builder builder = WinRmClient.builder(url, AuthSchemes.BASIC);
+
+        WinRmClient client = builder.build();
+
+        StringWriter out = new StringWriter();
+        StringWriter err = new StringWriter();
+        int code;
+
+        try {
+            code = client.command("echo myline", out, err);
+        } finally {
+            client.disconnect();
+        }
+
+        assertEquals(out.toString(), "myline\r\n");
+        assertEquals(err.toString(), "");
+        assertEquals(code, 0);
+    }
+
+    private static class RecordedSessionDispatcher extends Dispatcher {
+
+        String serverUrl;
+        String messageId;
+        Iterator<Item> requests;
+        String recordingName;
+        boolean hasErrors;
+
+        public RecordedSessionDispatcher(String recordingName, String serverUrl) {
+            this.serverUrl = serverUrl;
+            this.recordingName = recordingName;
+        }
+
+        public boolean hasErrors() {
+            return hasErrors;
+        }
+
+        @Override
+        public MockResponse dispatch(RecordedRequest request) throws InterruptedException {
+            try {
+                return doDispatch(request);
+            } catch (Throwable e) {
+                hasErrors = true;
+                if (e instanceof InterruptedException) {
+                    Thread.currentThread().interrupt();
+                    throw (InterruptedException)e;
+                }
+                LOG.log(Level.SEVERE, "Processing request " + request + " failed", e);
+                return new MockResponse().setResponseCode(500);
+            }
+        }
+
+        private MockResponse doDispatch(RecordedRequest request) {
+            Document requestDoc = parseRequest(request);
+            if (messageId == null) {
+                initRecordedSession(requestDoc);
+            }
+
+            Item next = requests.next();
+
+            String contentType = request.getHeader("Content-Type");
+            assertEquals(contentType, "application/soap+xml; action=\"" + next.getAction() + "\"; charset=UTF-8");
+            CompareMatcher.isIdenticalTo(next.getRequest())
+                .throwComparisonFailure()
+                .ignoreWhitespace()
+                .matches(requestDoc.getFirstChild());
+
+            Buffer response = serializeResponse(next.getResponse());
```

Comment:
What do you think about supplying it explicitly like here 
https://github.com/jclouds/jclouds/blob/master/providers/google-compute-engine/src/test/java/org/jclouds/googlecomputeengine/features/InstanceApiMockTest.java#L71
https://github.com/jclouds/jclouds/blob/master/providers/google-compute-engine/src/test/resources/instance_serial_port_4_windows.json